migration-engine/connectors/sql-migration-connector/src/flavour/mssql.rs

```diff
@@ -13,33 +15,31 @@ impl MssqlFlavour {
         self.0.schema()
     }
 
-    fn master_url(jdbc_string: &str) -> (String, String) {
-        let mut splitted = jdbc_string.split(';');
-        let uri = splitted.next().unwrap().to_string();
+    fn master_url(input: &str) -> ConnectorResult<(String, String)> {
+        let mut conn = JdbcString::from_str(input).map_err(|e| ConnectorError::generic(anyhow::Error::new(e)))?;
+        let server_name = conn.server_name().ok_or_else(|| {
+            ConnectorError::generic(format_err!("JDBC connection string did not contain a server name"))
+        })?;
 
-        let mut params: HashMap<String, String> = splitted
-            .map(|kv| kv.split('='))
-            .map(|mut kv| {
-                let key = kv.next().unwrap().to_string();
-                let value = kv.next().unwrap().to_string();
-
-                (key, value)
-            })
-            .collect();
+        let host = match conn.instance_name() {
+            Some(instance_name) => format!(r#"{}\{}"#, server_name, instance_name),
+            None => server_name.to_string(),
+        };
+        let params = conn.properties_mut();
 
         let db_name = params.remove("database").unwrap_or_else(|| String::from("master"));
         let params: Vec<_> = params.into_iter().map(|(k, v)| format!("{}={}", k, v)).collect();
```

Comment:
Do we handle the correct escaping for ADO.net in here? I mean is it the same as for JDBC?

Reply:
Ah, I wasn't quite sure how this worked but I see. How about instead of converting jdbc to ado.net inline here, we enabled `quaint` to take either. We could rewrite https://github.com/prisma/quaint/blob/7e310e21fff139bf797fb0307aa4a9fb56193bce/src/connector/mssql.rs#L225-L227 to instead be:

```rust
pub async fn new(url: MssqlUrl) -> crate::Result<Self> {
    let config = match Config::from_ado_string(&url.connection_string) {
        Ok(config) => config,
        Err(e) => Config::from_jdbc_string(&url.connection_string)?,
    };

    let tcp = TcpStream::connect_named(&config).await?;
```

That way we don't need to do the ad-hoc re-encoding here, but can instead make it so quaint can operate on either.

Reply:
Exactly. Sounds good.

Reply:
Oh I guess the `from_jdbc_string` was merged in https://github.com/prisma/tiberius/pull/92/. Let me update this PR to use that.